addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,237 @@ odoo.define('website_event.editor', function (require) {
+        options = _.defaults(options || {}, {
+            title: _t("New Event"),
+            size: 'medium',
+            buttons: [
+                {
+                    text: _t("Create"),
+                    classes: 'btn-primary',
+                    click: this._onClickCreate.bind(this),
+                },
+                {
+                    text: _t("Discard"),
+                    close: true
+                },
+            ]
+        });
+        this.eventStart = moment();
+        this.eventEnd = moment().add(1, "d");
+        this._super(parent, options);
+    },
+
+    /**
+     * @override
+     */
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$('input.daterange-input').each(function () {
+                self._initDateRangePicker($(this));
+            });
+
+            var $addressInput = self.$('#address_ids');
+            $addressInput.select2({
+                width: '100%',
+                allowClear: true,
+                formatNoMatches: false,
+                multiple: false,
+                selection_data: false,
+                formatSelection: function (data) {
+                    if (data.name) {
+                        data.text = data.name;
+                    }
+                    return data.name;
+                },
+                createSearchChoice: function (name, data) {
+                    var addedPartner = $(this.opts.element).select2('data');
+                    if (_.filter(_.union(addedPartner, data), function (partner) {
+                        return partner.text.toLowerCase().localeCompare(name.toLowerCase()) === 0;
+                    }).length === 0) {
+                        return {
+                            id: _.uniqueId('address_'),
+                            create: true,
+                            text: _.str.sprintf(_t('Create "%s"'), name),
+                            name: name,
+                        };
+                    }
+                },
+                fill_data: function (query, data) {
+                    var that = this;
+                    var partners = {results: []};
+                    _.each(data, function (obj) {
+                        var contact_address = obj.contact_address.replace(/[\n]+/g, '').trim();
+                        if (contact_address && that.matcher(query.term, contact_address)) {
+                            partners.results.push({id: obj.id, text: contact_address});
+                        }
+                    });
+                    query.callback(partners);
+                },
+                query: function (query) {
+                    var that = this;
+                    // fetch data only once and store it
+                    if (!this.selection_data) {
+                        self._rpc({
+                            model: 'res.partner',
+                            method: 'search_read',
```

Comment:
Are we doing a search_read without any other parameter than the fields to read ? I mean are we going to read our 3M partners ?